src/amo/reducers/shelves.js

```diff
@@ -123,6 +128,26 @@ const reducer = (
       };
     }
 
+    // See: https://github.com/mozilla/addons-frontend/issues/8601
+    case LOCATION_CHANGE: {
+      if (_config.get('server')) {
+        // We only care about client side navigation.
+        return state;
+      }
+      // When the client initializes, it updates its location. On next location
+      // change, we want to reset this state to fetch fresh data once user goes
+      // back to the homepage.
+      if (state.resetStateOnNextChange) {
+        return initialState;
+      }
+
+      return {
+        ...state,
+        // This will only be set *after* a single location change on the client.
+        resetStateOnNextChange: true,
+      };
```

Comment:
Heh. I don't like this hack. I know we have to keep it but.. still.

Reply:
I agree. We now have it in 3 places, but it seems to work and we haven't come up with anything better.